plugins/net_plugin/net_plugin.cpp

```diff
@@ -4748,21 +4737,31 @@ namespace eosio {
    }
 
    std::optional<connection_status> connections_manager::status( const string& host )const {
-      std::shared_lock g( connections_mtx );
-      auto con = find_connection_i( host );
+      connection_ptr con;
+      {
+         std::shared_lock g( connections_mtx );
+         con = find_connection_i( host );
+      }
```

Comment:
Should we keep the lock until we are done using `con`?

Reply:
No. `get_status` grabs a different mutex. Safer to not hold both.

Reply:
So should we do the same [here](https://github.com/AntelopeIO/spring/blob/6d36988895b0b2d148e15e32a801b17cff55fd6f/plugins/net_plugin/net_plugin.cpp#L4741)

Reply:
I did do it there.